File: typescript/src/transport.ts
/**
 * Best-effort JSON POST. Telemetry/platform writes must never break the
 * originating request, so failures are logged and swallowed (returns null).
 */

export async function postJson(
  url: string,
  payload: unknown,
  opts: { headers?: Record<string, string>; timeoutMs?: number; label?: string } = {},
): Promise<Response | null> {
  const label = opts.label ?? "platform";
  const controller = new AbortController();
  const timer = setTimeout(() => controller.abort(), opts.timeoutMs ?? 5000);
  try {
    const resp = await fetch(url, {
      method: "POST",
      headers: { "content-type": "application/json", ...(opts.headers ?? {}) },
      body: JSON.stringify(payload),
      signal: controller.signal,
    });
    if (!resp.ok) {
      console.warn(`[streamoid:${label}] POST ${url} -> ${resp.status}`);
      return null;
    }
    return resp;
  } catch (err) {
    console.warn(`[streamoid:${label}] POST ${url} failed:`, (err as Error)?.message);
    return null;
  } finally {
    clearTimeout(timer);
  }
}

export function authHeaders(token?: string): Record<string, string> {
  return token ? { authorization: `Bearer ${token}` } : {};
}

File: python/src/streamoid/transport.py
"""Best-effort async HTTP helper shared by the SDK clients.

Every platform write (event, audit row, file-index row, MCP registration) is
**non-critical to the originating request**: if the platform plane is briefly
unreachable, the user's generation/upload must still succeed. So this helper
swallows and logs transport errors and returns ``None`` instead of raising.
Callers that need delivery guarantees should layer a queue on top later; for
the vendored phase, log-on-failure matches the architecture's "frontend events
may be lossy; server events are emitted but must not take the request down".
"""

from __future__ import annotations

import logging
from typing import Any, Dict, Optional

import httpx

logger = logging.getLogger(__name__)


async def post_json(
    url: str,
    payload: Dict[str, Any],
    *,
    headers: Optional[Dict[str, str]] = None,
    timeout_s: float = 5.0,
    label: str = "platform",
) -> Optional[httpx.Response]:
    try:
        async with httpx.AsyncClient(timeout=timeout_s) as client:
            resp = await client.post(url, json=payload, headers=headers or {})
        if resp.status_code >= 400:
            logger.warning(
                "%s POST %s -> %s: %s", label, url, resp.status_code, resp.text[:300]
            )
            return None
        return resp
    except httpx.HTTPError as exc:
        logger.warning("%s POST %s failed: %s", label, url, exc)
        return None
    except Exception as exc:  # never let a telemetry write escape into the request
        logger.warning("%s POST %s unexpected error: %s", label, url, exc)
        return None


def auth_headers(token: Optional[str]) -> Dict[str, str]:
    return {"Authorization": f"Bearer {token}"} if token else {}
